kubernetes/samples/scripts/create-weblogic-domain/model-in-image/build-model-image.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+# Copyright (c) 2019, 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+# 
+#  Summary:
+#
+#    This script builds a model image using the WebLogic Image Tool. The
+#    tool pulls a base image if there isn't already a local base image.
+#    This script, by default, builds the model image with model files from
+#    './stage-model-image.sh' plus tooling downloaded by './stage-tooling.sh.'.
+#
+#  Assumptions:
+#
+#    - The WebLogic Image Tool zip is 'WORKDIR/weblogic-image-tool.zip' and
+#      the WebLogic Deploy Tool zip is 'WORKDIR/weblogic-deploy-tooling.zip'
+#      (see './stage-tooling.sh').
+#
+#    - Model files have been staged in the MODEL_DIR directory
+#      (by './stage-model-image.sh' or some custom process).
+#
+#  Optional environment variables:
+#
+#    WORKDIR
+#      Working directory for the sample with at least 10GB of space.
+#      Defaults to '/tmp/$USER/model-in-image-sample-work-dir'.
+#
+#    MODEL_DIR:
+#      Location of the model .zip, .properties, and .yaml files
+#      that will be copied to the model image.  Default is:
+#        'WORKDIR/model/image--$(basename $MODEL_IMAGE_NAME):$MODEL_IMAGE_TAG'
+#      which is usually populated by the './stage-model-image.sh' script.
+#
+#    MODEL_IMAGE_BUILD:
+#      Set to 'when-changed' (default) or 'always'. Default behavior is to
+#      exit without building if the docker image BASE_IMAGE_NAME:BASE_IMAGE_TAG
+#      is found in the local docker image cache.
+#
+#    WDT_DOMAIN_TYPE
+#      'WLS' (default), 'RestrictedJRF', or 'JRF'.
+#
+#    BASE_IMAGE_NAME, BASE_IMAGE_TAG:
+#      The base image name defaults to 
+#         'container-registry.oracle.com/middleware/weblogic'
+#      for the 'WLS' domain type, and otherwise defaults to 
+#         'container-registry.oracle.com/middleware/fmw-infrastructure'. 
+#      The tag defaults to '12.2.1.4'.
+#
+#    MODEL_IMAGE_NAME, MODEL_IMAGE_TAG:
+#      Defaults to 'model-in-image' and 'v1'.
+#
+
+set -eu
+set -o pipefail
+
+SCRIPTDIR="$( cd "$(dirname "$0")" > /dev/null 2>&1 ; pwd -P )"
+source $SCRIPTDIR/env-init.sh
+
+cd ${WORKDIR}
+
+echo @@ Info: WDT_DOMAIN_TYPE=${WDT_DOMAIN_TYPE}
+echo @@ Info: MODEL_DIR=${MODEL_DIR}
+echo @@ Info: BASE_IMAGE_NAME=${BASE_IMAGE_NAME}
+echo @@ Info: BASE_IMAGE_TAG=${BASE_IMAGE_TAG}
+echo @@ Info: MODEL_IMAGE_NAME=${MODEL_IMAGE_NAME}
+echo @@ Info: MODEL_IMAGE_TAG=${MODEL_IMAGE_TAG}
+echo @@ Info: MODEL_IMAGE_BUILD=${MODEL_IMAGE_BUILD}
+
+if [ ! "$MODEL_IMAGE_BUILD" = "always" ] \
```

Comment:
This feature just doesn't make sense.  If I call a script called `build-model-image.sh`, then it should build the image.

Reply:
This was requested by Mark.  This script is intended to be closer to 'production' than 'sample' - so it has 'optimizations' like this.  One side effect is that it forces the user to think once they see the message, to see if there intent was to obliterate an existing image.  For example, maybe they need to change the image name/tag.